common/scala/src/main/scala/whisk/core/containerpool/HttpUtils.scala

```diff
@@ -51,9 +56,11 @@ import whisk.core.entity.size.SizeLong
  * @param hostname the host name
  * @param timeout the timeout in msecs to wait for a response
  * @param maxResponse the maximum size in bytes the connection will accept
+ * @param maxConcurrent the maximum number of concurrent requests allowed
  */
-protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize) {
-
+protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize, maxConcurrent: Int) {
+  def this(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize) =
+    this(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize, 1)
```

Comment:
Could you not use a default parameter for `maxConcurrent` here?

Reply:
This was a problem for the compiler - test classes become unable to use `object HttpUtils`, even though it is not protected.

Reply:
lol @markusthoemmes and i made the same comment i didn't see it sorry.